docs/source/1.0/guides/migrationg-idl-1-to-2.rst

```diff
@@ -0,0 +1,292 @@
+=====================================
+Smithy IDL 1.0 to 2.0 Migration Guide
+=====================================
+
+This guide describes how to migrate your models from Smithy IDL version 1.0
+to version 2.0 without breaking your models or customers.
+
+.. contents:: Table of contents
+    :depth: 2
+    :local:
+    :backlinks: none
+
+Update the model file version
+=============================
+
+For each model file you are upgrading, change the version from ``1`` or
+``1.0`` to ``2.0``. In the IDL this is controlled with the
+:ref:`version statement <smithy-version>`, and in the AST it is controlled
+with the ``smithy`` :ref:`top level property <top-level-properties>`. For
+example, the following model:
+
+.. code-block:: smithy
+
+    $version: "1.0"
+
+    namespace smithy.example
+
+    string Foo
+
+Should be updated to:
+
+.. code-block:: smithy
+
+    $version: "2.0"
+
+    namespace smithy.example
+
+    string Foo
+
+An IDL model file may not have a version control statement at the top. In this
+case, it uses version 1.0 by default. For example:
+
+.. code-block:: smithy
+
+   namespace smithy.example
+
+   string Foo
+
+Also needs to be updated to:
+
+.. code-block:: smithy
+
+    $version: "2.0"
+
+    namespace smithy.example
+
+    string Foo
+
+Smithy's tooling is able to load both 1.0 model files and 2.0 model files into
+a single combined model. Therefore when migrating, it may be wise to migrate
+one file at a time.
+
+
+Remove the box trait
+====================
+
+The ``box`` trait is removed in 2.0, so it must be removed from any shapes or
+members that use it. Smithy structure members are considered boxed by default,
+which can be changed using the :ref:`required-trait` or :ref:`default-trait`.
+
+.. seealso::
+
+    :ref:`optionality`
+
+
+Replace Primitive prelude shape targets
+=======================================
+
+The primitive shapes have been removed from the prelude, and so any member
+targeting one of them must update to target its equivalent non-primitive
+shape.
+
+.. list-table:
```

Comment:
Whe updating structure members that target these, they'll need to add the `@default` trait.